gui/webdisplay/src/TWebWindowsManager.cxx

```diff
@@ -593,29 +617,58 @@ bool ROOT::Experimental::TWebWindowsManager::Show(ROOT::Experimental::TWebWindow
       is_firefox = prog.Length() > 0;
 
 #ifdef _MSC_VER
-      if (win.IsBatchMode())
+      if (batch_mode)
          // there is a problem when specifying the window size with wmic on windows:
          // It gives: Invalid format. Hint: <paramlist> = <param> [, <paramlist>].
-         exec = gEnv->GetValue("WebGui.FirefoxBatch", "fork: -headless -no-remote $url");
+         exec = gEnv->GetValue("WebGui.FirefoxBatch", "fork: -headless -no-remote $profile $url");
       else
-         exec = gEnv->GetValue("WebGui.FirefoxInteractive", "$prog -width=$width -height=$height $url");
+         exec = gEnv->GetValue("WebGui.FirefoxInteractive", "$prog -width=$width -height=$height $profile $url");
 #else
-      if (win.IsBatchMode())
-         exec = gEnv->GetValue("WebGui.FirefoxBatch", "fork:-headless -no-remote $url");
+      if (batch_mode)
+         exec = gEnv->GetValue("WebGui.FirefoxBatch", "fork:-headless -no-remote $profile $url");
       else
-         exec = gEnv->GetValue("WebGui.FirefoxInteractive", "$prog -width $width -height $height \'$url\' &");
+         exec = gEnv->GetValue("WebGui.FirefoxInteractive", "$prog -width $width -height $height $profile \'$url\' &");
 #endif
+
+      if (is_firefox && (exec.Index("$profile") != kNPOS)) {
+         TString repl;
```

Comment:
This is the string that substitutes "$profile", right? Then maybe rename this variable to "profileValue"? "repl" is very generic.

Reply:
I call it `profile_arg` - it is profile argument of firefox launcher